src/librustdoc/html/render/print_item.rs

```diff
@@ -376,17 +376,26 @@ fn item_module(w: &mut Buffer, cx: &Context<'_>, item: &clean::Item, items: &[cl
                 let stab = myitem.stability_class(cx.tcx());
                 let add = if stab.is_some() { " " } else { "" };
 
+                let visibility_emoji = match myitem.visibility {
+                    clean::Visibility::Restricted(_) => {
+                        "<span title=\"Restricted Visibility\">&nbsp;🔒</span> "
```

Comment:
@camelid pointed out that “Private” is a bit ambiguous in this case:

https://github.com/rust-lang/rust/pull/95024#issuecomment-1071171332

The icon/emoji also appears for `pub(...)` items, hence, I think “Restricted Visibility” is more precise. However, I am also fine with “Private” and happy to modify the PR accordingly. :)

Reply:
Thanks for pointing that out.

> Also, the definition privacy for these purposes is ambiguous: does it mean private to the module you're looking at or private to the whole crate?

From the perspective of API design, "private" means "private from a view outside the crate." In other words, it includes everything except what's marked with plain `pub`; it excludes `pub(crate)`, `pub(self)`, etc. Since rustdoc is about APIs, I think it makes sense to use that definition of private. @camelid what do you think?

Reply:
An example may help here. Is the struct `A` in the following snippet *private*?
```rust
mod(crate) private {
    pub struct A; // <-- Is this private?

    struct B;
}
```
It cannot be accessed from “outside the crate” because it is in a private module. However, the struct itself is `pub` and will thus not be marked with a 🔒 with the present change. (This also makes sense if we think about designing an API to be used by other modules in the crate which is one of my use cases for this. From the perspective of other modules in the same crate it is indeed public unlike struct `B`.)

Reply:
I feel like the `A` struct should be marked as 🔒 since as you said earlier, the 🔒 is about whether it's accessible to code outside the crate. Perhaps that's too technically difficult to implement though?

Reply:
Yes, this seems difficult to implement, at least for me. I do not know how to extract from the compiler whether `A` is somehow publicly accessible or not. Does the compiler have this information at all? I also think it makes sense to think about a `pub` item in a private module about being public (after all it is explicitly declared public) but *with respect to* an internal API available to other modules. It is that *internal API* that is documented. While I agree that Rustdoc is about APIs, I like to challenge that it is primarily about the APIs of crates. Documenting private items and crate internals can be highly valuable, e.g., for bringing contributors up-to-speed or to document the internal architecture and design decision behind it. This is my/our use case for this change.

Anyway, what does this discussion entail about the “Restricted Visibility” label? Should that be kept as it currently is or should I change it to “Private”? All items marked that will be marked with 🔒 by the current change are private in the sense that they cannot be accessed from outside the crate.

Reply:
`std::intrinsics::transmute` isn't marked as unstable despite `std::intrinsics` being marked as such. It is re-exported as `std::mem::transmute`. I think the restricted visibility docs should behave the same way. If you l

Reply:
Good point, I hadn't thought of that.

Reply:
There is a query for this, it's named `is_publicly_accessible` or something similar. It's used for unreachable_pub.